Allow to select the edit layout for subforms in custom fields

HLeithner opened the pull request:
### Summary of Changes
Add a column to the subform custom field which allows to select they layout for editing


### Testing Instructions
* create a subform
* select the type of the edit layout
* layout is used on editing


### Actual result BEFORE applying this Pull Request
Not possible


### Expected result AFTER applying this Pull Request
Works


### Documentation Changes Required
[possible
](https://docs.joomla.org/Help4.x:Fields:_Edit)

HLeithner committed:
Allow to select the edit layout for subforms in custom fields

brianteeman commented:
This is a good example of issues with the editorconfig that we now have.

The file is using tabs but your new code is using spaces :(

HLeithner commented:
> This is a good example of issues with the editorconfig that we now have.
> 
> The file is using tabs but your new code is using spaces :(

hmm you are right, seems I was to optimistic for the editorconfig.

I will write a PR that only php files uses spaces.

HLeithner committed:
Merge branch '4.2-dev' of github.com:joomla/joomla-cms into feature/subform-editlayout

HLeithner committed:
cs

HLeithner commented:
Now I used tabs in the xml

Quy commented on administrator/language/en-GB/plg_fields_subform.ini:
```diff
@@ -10,4 +10,7 @@ PLG_FIELDS_SUBFORM_PARAMS_MAX_ROWS_LABEL="Maximum Rows"
 PLG_FIELDS_SUBFORM_PARAMS_OPTIONS_LABEL="Fields"
 PLG_FIELDS_SUBFORM_PARAMS_RENDER_VALUES_LABEL="Render Values"
 PLG_FIELDS_SUBFORM_PARAMS_REPEAT_LABEL="Repeatable"
+PLG_FIELDS_SUBFORM_PARAMS_EDIT_LAYOUT_LABEL="Edit Layout"
+PLG_FIELDS_SUBFORM_PARAMS_EDIT_LAYOUT_OPTION_REPEATABLE_TABLE_LABEL="Repeatable Table"
+PLG_FIELDS_SUBFORM_PARAMS_EDIT_LAYOUT_OPTION_REPEATABLE_FORM_LABEL="Repeatable Form"
```
Please alpha sort keys.

Hackwar replied:
Done

crystalenka commented:
Works well, and is a need! Just would recommend moving the field to Options tab in the Form Options field group, as that pattern is already established in other field types like radio:

<img width="1512" alt="" src="https://user-images.githubusercontent.com/9141288/198612345-c8bd4e2c-0646-42fe-89be-0fa7c0c5767a.png">

crystalenka committed:
Minor updates to field location

Making the field consistent with the existing pattern of other form layout selects (list, radio, etc)

Only shows the select when field is repeatable, and removes redundant 'repeatable' word from language strings.

HLeithner committed:
Merge pull request #50 from crystalenka/subform-layout-select

Minor updates to field location

crystalenka commented:
Hahaha my change made something not work. Sorry. 🙈 I'll open another PR against your branch after Amelia is in bed and I can fix it.

crystalenka committed:
Fixing typo

Makes it work as intended now. Oops!

HLeithner committed:
Merge pull request #51 from crystalenka/feature/subform-layout-select

Fixing typo

Hackwar committed:
Merge branch '4.3-dev' into feature/subform-editlayout

Hackwar commented:
Since 4.2 only accepts Bugfixes and this is a new feature (or rather an improvement), I've rebased the PR to 4.3-dev and updated it.

Hackwar committed:
alpha ordering strings

QuyTon committed:
Merge branch '4.3-dev' into feature/subform-editlayout

Quy commented:
I have tested this item :white_check_mark: successfully on f25d7594556e945d5b3e553401fbd4b35e777e8a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38180">issues.joomla.org/tracker/joomla-cms/38180</a>.</sub>

carlitorweb commented:
I have tested this item :white_check_mark: successfully on f25d7594556e945d5b3e553401fbd4b35e777e8a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38180">issues.joomla.org/tracker/joomla-cms/38180</a>.</sub>

carlitorweb commented:
Documentation added :heavy_check_mark: 
https://docs.joomla.org/Help4.x:Fields:_Edit

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38180">issues.joomla.org/tracker/joomla-cms/38180</a>.</sub>

QuyTon committed:
Merge branch '4.3-dev' into feature/subform-editlayout

HLeithner commented:
Thanks @carlitorweb for the documentation update
